pageserver/src/consumption_metrics.rs

```diff
@@ -422,6 +417,102 @@ async fn collect_metrics_iteration(
     }
 }
 
+async fn collect<S>(
```

Comment:
This was at the wrong level and I did not end up needing it, I can inline it at the end. Doing it now would just be conflict after conflict.